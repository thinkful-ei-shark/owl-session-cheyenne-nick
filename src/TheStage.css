.the-stage .stage-screen {
    position: relative;
    top: 330px;
    left: -50%;
}