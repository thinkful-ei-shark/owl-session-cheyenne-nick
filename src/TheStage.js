import React, { Component } from 'react';
import './TheStage.css'
import StageParticipant from './StageParticipant.js'

class TheStage extends Component {
    render(){
        const participants = this.props.participants.map(participant =>{
            return < StageParticipant key = {participant.id} name = {participant.name} avatar = {participant.avatar}/>
        });

        return(
           <div className="the-stage three-column">
               <div className="stage-paricipants group-row">
                   {participants}
                   <div className = "stage-screen">
                       <div className = "audio-bar"></div>
                       <img  src={this.props.avatar} alt={this.props.name} className="profile-pic item" />

                   </div>
               </div>
           </div>
        )
    }
}

export default TheStage;
