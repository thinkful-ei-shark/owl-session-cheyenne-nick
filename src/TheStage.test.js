import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TheStage from './TheStage'

describe('<TheStage />', () => {
    it('renders without crashing', () => {
        // Create a DOM element to render the component into
        const div = document.createElement('div');
        const participants = [{
          id: 1,
          name: 'Koren Templeton',
          avatar:
            'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: true
        },
        {
          id: 2,
          name: 'Caty Flucker',
          avatar:
            'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
          inSession: true,
          onStage: false
        }]

        // Render the component
        // If something is wrong it will fail here
        ReactDOM.render(<TheStage participants={participants} avatar={participants[0].avatar} 
          name={participants[0].name} />, div);

        // Clean up
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
      const participants = [{
        id: 1,
        name: 'Koren Templeton',
        avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: true
      },
      {
        id: 2,
        name: 'Caty Flucker',
        avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
        inSession: true,
        onStage: false
      }];

      const stage = renderer
          .create(<TheStage participants={participants} avatar={participants[0].avatar} name={participants[0].name} />)
          .toJSON();
      
      expect(stage).toMatchSnapshot();
    });
});
