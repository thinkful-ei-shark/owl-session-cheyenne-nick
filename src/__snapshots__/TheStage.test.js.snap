// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TheStage /> renders the UI as expected 1`] = `
<div
  className="the-stage three-column"
>
  <div
    className="stage-paricipants group-row"
  >
    <div
      className="stage-participant item group-column"
    >
      <div
        className="stage-settings item group-column"
      >
        <h3
          className="stage-name"
        >
          this.props.name
        </h3>
        <div
          className="audio-bar"
        />
        <div
          className="icons group-row"
        >
          <button
            className="speaker"
          >
            🔊
          </button>
          <button
            className="mag-glass"
          >
            🔍
          </button>
        </div>
      </div>
      <img
        alt="Koren Templeton"
        className="profile-pic item"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="stage-participant item group-column"
    >
      <div
        className="stage-settings item group-column"
      >
        <h3
          className="stage-name"
        >
          this.props.name
        </h3>
        <div
          className="audio-bar"
        />
        <div
          className="icons group-row"
        >
          <button
            className="speaker"
          >
            🔊
          </button>
          <button
            className="mag-glass"
          >
            🔍
          </button>
        </div>
      </div>
      <img
        alt="Caty Flucker"
        className="profile-pic item"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="stage-screen"
    >
      <div
        className="audio-bar"
      />
      <img
        alt="Koren Templeton"
        className="profile-pic item"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
</div>
`;
